src/utils_general/ghs.py

```diff
@@ -0,0 +1,120 @@
+import glob
+import rasterio
+from rasterio.merge import merge
+from utils_general import utils
+import logging
+from pathlib import Path
+import os
+from typing import List, Tuple, Union
+import numpy as np
+from rasterstats import zonal_stats
+
+logger = logging.getLogger(__name__)
+
+
+def get_ghs_data(
+    ghs_type: str,
+    column_row_pairs: List[Tuple[int, int]],
+    country_iso3: str,
+    output_dir: Union[str, Path],
+):
+    """
+    Download GHS data from the JRC portal.
+
+    :param ghs_type: One of either "SMOD" or "POP".
+    :param column_row_pairs: List of column & row tuples specifying
+        tiles to download.
+    :param country_iso3: Country ISO3 code.
+    :param output_dir: Path or string specifying output directory
+        for output saving.
+
+    :return: Mosaic raster
+    """
+
+    # URL paths for GHS download
+
+    URL_BASE = "https://cidportal.jrc.ec.europa.eu/ftp/jrc-opendata/GHSL"
+    GHS_URL = {
+        "SMOD": "GHS_SMOD_POP_GLOBE_R2019A"
+        "/GHS_SMOD_POP2015_GLOBE_R2019A_54009_1K/V2-0/tiles/"
+        "GHS_SMOD_POP2015_GLOBE_R2019A_54009_1K_V2_0_{column}_{row}.zip",
+        "POP": "GHS_POP_MT_GLOBE_R2019A"
+        "/GHS_POP_E2015_GLOBE_R2019A_54009_1K/V1-0/tiles/"
+        "GHS_POP_E2015_GLOBE_R2019A_54009_1K_V1_0_{column}_{row}.zip",
+    }
+
+    # Set filepaths
+    download_dir = os.path.join(output_dir, "ghs", "zip")
+    output_filepath = os.path.join(
+        output_dir,
+        "ghs",
+        "{country_iso3}_{ghs_type}_2015_1km_mosaic.tif".format(
```

Comment:
Is it correct that this data doesn't get updated? If so, would it make sense to use an `use_cache` boolean. Such that it doesn't automatically redownload data when data already exists? See e.g. [here](https://github.com/OCHA-DAP/pa-anticipatory-action/blob/a1fefb4d34124393c87d50b0214887ae4dc36903/src/indicators/flooding/glofas/glofas.py#L87)

Reply:
That is a very good idea!